backend/src/layer/data/database/mealplan_management.rs

```diff
@@ -0,0 +1,842 @@
+use async_trait::async_trait;
+use sqlx::{Pool, Postgres};
+
+use super::types::DatabasePrice;
+use crate::{
+    interface::persistent_data::{MealplanManagementDataAccess, Result},
+    util::{Additive, Allergen, Date, MealType, Price, Uuid},
+};
+
+pub struct PersistentMealplanManagementData {
+    pub(super) pool: Pool<Postgres>,
+}
+
+const THRESHOLD: f32 = 0.785;
+
+#[async_trait]
+#[allow(clippy::missing_panics_doc)] // necessary because sqlx macro sometimes create unreachable panics?
+impl MealplanManagementDataAccess for PersistentMealplanManagementData {
+    async fn dissolve_relations(&self, canteen_id: Uuid, date: Date) -> Result<()> {
+        sqlx::query!(
+            "
+            DELETE FROM food_plan
+            WHERE serve_date = $1
+            AND line_id IN (SELECT line_id FROM line WHERE canteen_id = $2)
+            ",
+            date,
+            canteen_id
+        )
+        .execute(&self.pool)
+        .await?;
+        Ok(())
+    }
+
+    async fn get_similar_canteen(&self, similar_name: &str) -> Result<Option<Uuid>> {
```

Comment:
First, it is impossible that there are 3 similar meals, because this function exists. But if there are three equal canteens, the database decides which canteen will be in the first row. And as there is a system beneath, the canteen will always be the same.